src/reporter/tests/test_NTNENA.py

```diff
@@ -1189,3 +1190,213 @@ def test_aggregation_is_per_instance(service, reporter_dataset):
+    assert isinstance(obtained, dict)
+    expected_attrs = [
+        {
+            'attrName': 'boolean',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': [True]
+                }],
+                'entityType': 'Test'
+            }]
+        },
+        {
+            'attrName': 'intensity',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': ['Low']
+                }],
+                'entityType': 'Test'
+            }]
+        },
+        {
+            'attrName': 'temperature',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': [10.0]
+                }],
+                'entityType': 'Test'
+            }]
+        },
+    ]
+
+    expected = {
+        'attrs': expected_attrs
+    }
+    obtained = r.json()
+    assert obtained == expected
+
+    # 'aggrMethod': 'count'
+
+    query_params = {
+        'attrs': attrs,
+        'aggrMethod': 'count',
+    }
+    r = query(params=query_params, service=service)
+    assert r.status_code == 200, r.text
+
+    obtained = r.json()
+    assert isinstance(obtained, dict)
+    expected_attrs = [
+        {
+            'attrName': 'boolean',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': [1]
+                }],
+                'entityType': 'Test'
+            }]
+        },
+        {
+            'attrName': 'intensity',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': [1]
+                }],
+                'entityType': 'Test'
+            }]
+        },
+        {
+            'attrName': 'temperature',
+            'types':
+            [{
+                'entities':
+                [{
+                    'entityId': 'Test1',
+                    'index': ['',''],
+                    'values': [1]
+                }],
+                'entityType': 'Test'
+            }]
+        },
+    ]
+
+    expected = {
+        'attrs': expected_attrs
+    }
+
+    obtained = r.json()
+    assert obtained == expected
+
+    # 'aggrMethod': 'avg'
+
+    query_params = {
+        'attrs': attrs,
+        'aggrMethod': 'avg'
+    }
+    r = query(params=query_params, service=service)
+    assert r.status_code == 404, r.text
+    assert r.json() == {
+        "error": "Not Found",
+        "description": "No records were found for such query."
+    }
+
```

Comment:
while 404 is indeed what to expect, i wonder if there is a way to have more information in the description. e.g. avg does not work on text and boolean attributes?

Reply:
surfacing the issue, may not be easy, but at least we can log it if the db says anything at all:
https://github.com/orchestracities/ngsi-timeseries-api/blob/35d3b93dde3189d95a4bf8f29a1ae178368a0606/src/translators/sql_translator.py#L1110

Reply:
I am looking into it and update once it is fixed.